### ISSUE TITLE
No easy way to determine if a method is partial.

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#6956]
From my inbox...

---

How to query if a method is a partial definition or partial implementation? 

``` cs
partial class Example 
{
partial void Test();
}
```

I’ve tried `PartialDefinitionPart` and `PartialImplementationPart`, but it seems `PartialDefinitionPart` always return null, even for Test() above.

`method.PartialDefinitionPart` returns null if the method itself is the definition.
Likewise `PartialImplementationPart` returns null if the method itself is the implementation.

``` cs
/// 
/// If this is a partial method implementation part, returns the corresponding
/// definition part. Otherwise null.
/// 
IMethodSymbol PartialDefinitionPart { get; }

/// 
/// If this is a partial method declaration without a body, and the method is
/// implemented with a body, returns that implementing definition. Otherwise
/// null.
/// 
IMethodSymbol PartialImplementationPart { get; }
```

There seems to be something missing. There doesn’t seem to be any way to know if a method is partial _unless_ the other part happens to exist. i.e. if I have a partial definition, with no partial implementation, 
How would I find out that I’m actually partial?

Well, it is a bit hacky, but you may be able to find this out by `method.CallsAreOmitted(tree) && !method.CallsAreConditionallyOmitted(tree)`. However, it probably makes sense to expose `IsPartialDefinition`.

Or, perhaps the simplest change would be to **have `PartialDefinitionPart` return itself for the definition part, and `PartialImplementationPart` return itself for the implementation part**.

**Comment by aetos382** [comment]
Is this issue resolved by #51255 ?

**Comment by CyrusNajmabadi** [comment]
Yes. t his can be closed out.  Thanks!
